starter-code/src/components/ContactsList.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from "react";
+import Contact from "./Contact";
+
+class ContactsList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      contacts: props.contacts.splice(0, 5),
+      sort: "none"
+    };
+  }
+
+  deleteContactHandler = contactIndex => {
+    const contactsCopy = this.state.contacts;
+    contactsCopy.splice(contactIndex, 1);
+    this.setState({
+      contacts: contactsCopy
+    });
+  };
+
+  addContact = () => {
+    let rand = Math.floor(Math.random() * this.props.contacts.length);
+    let contact = this.props.contacts[rand];
+    const contactsNew = this.state.contacts;
```

Comment:
This is a copy by reference. Prefer copying the values
```suggestion
    const contactsNew = [...this.state.contacts]; // or .slice()
   
```